src/devices/src/virtio/console/mod.rs

```diff
@@ -0,0 +1,56 @@
+
+mod device;
+mod inorder_handler;
+mod queue_handler;
+
+pub use device::Console;
+
+use crate::virtio::features::{VIRTIO_F_VERSION_1};
+
+// Console device ID as defined by the standard.
+pub const CONSOLE_DEVICE_ID: u32 = 3;
+
+// Console device resize feature.
+pub const VIRTIO_CONSOLE_F_SIZE: u32 = 0;
+
+// Console device multiport feature.
+pub const VIRTIO_CONSOLE_F_MULTIPORT: u32 = 1;
+
+// Console device emergency write feature.
+pub const VIRTIO_CONSOLE_F_EMERG_WRITE: u32 = 2;
+
+
+#[derive(Debug)]
+pub enum Error {
+    Virtio(crate::virtio::Error),
+}
+pub type Result<T> = std::result::Result<T, Error>;
+
+// Arguments required when building a console device.
+pub struct ConsoleArgs {
+    // pub allow_resize: bool,
+    // pub allow_multiport: bool,
+    // pub allow_emerg_write: bool,
+}
+
+impl ConsoleArgs {
+    // Generate device features based on the configuration options.
+    pub fn device_features(&self) -> u64 {
+        
+        let features = 1 << VIRTIO_F_VERSION_1;
```

Comment:
I think we could enable VIRTIO_F_IN_ORDER, though the console already handles the descriptors in order. I do not know if we need the event suppression of the VIRTIO_F_RING_EVENT_IDX since we want to be notified every time.

Reply:
Synced in private on this. Resolving it.